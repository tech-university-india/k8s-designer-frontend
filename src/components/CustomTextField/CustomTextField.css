.custom-textfield-container{
    display: flex;
    flex-direction: row;
    justify-content: space-between!important;
    gap: 0px 10px;
    position: relative;
}
.custom-textfield>*{
   margin: 8px 0px 0px 0px!important;
}
.add-textfield-btn{
    position: absolute;
    left:300px;
    bottom: 245px;
    color: rgb(19, 51, 124)!important;
}

@media screen and (max-width:768px) {
    .add-textfield-btn{
        left:155px;
        bottom:270px
    }
}