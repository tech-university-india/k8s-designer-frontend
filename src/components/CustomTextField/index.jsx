import * as React from 'react';
import './CustomTextField.css';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import PropTypes from 'prop-types';

export default function CustomTextField({userConfig,handleUserConfig,setNewTextField}){

  const [newConfigKey , setNewConfigKey] = React.useState();
  const [newConfigValue , setNewConfigValue] = React.useState();

  React.useEffect(() => {
    return ()=>{
      console.log('new config added');

      if(newConfigKey && newConfigValue)
      {
        console.log('new config added');
        userConfig = {...userConfig,[newConfigKey]:newConfigValue};
        handleUserConfig(userConfig);
        setNewTextField([]);
      }
    };
  }, []);

  return(
    <>
      <Box
        component="form"
        sx={{
          '& > :not(style)': { m: 1, width: '9.1vw' },
        }}
        noValidate
        autoComplete="off"
        className='custom-textfield-container config-layout'
        style = {
          { fontFamily :  'Open Sans'}
        }
      >
        <TextField id="outlined-basic" label = "Enter Key" variant="outlined" data-testid = "newConfig-key" value={newConfigKey} onChange={(event)=>{setNewConfigKey(event.target.value);}}/>
        <TextField id="outlined-basic" label = "Enter Value" data-testid = "newConfig-value" value={newConfigValue} variant="outlined" onChange={(event)=>{setNewConfigValue(event.target.value);}}/>
      </Box>
    </>
  );
}

CustomTextField.propTypes = {
  userConfig: PropTypes.object,
  handleUserConfig: PropTypes.func,
  setNewTextField: PropTypes.func
};