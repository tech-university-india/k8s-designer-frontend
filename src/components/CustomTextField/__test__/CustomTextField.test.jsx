import {fireEvent, render ,screen } from '@testing-library/react';
import CustomTextField from '..';
import * as React from 'react';

describe('Frontend Configuration',()=>{

  const mock = {
    userConfig:{
      PORT:'',
      NUMBER_OF_REPLICAS : '',
      ENVIRONMENT:'',
      DATABASE_URL:''
    },
    handleUserConfig:jest.fn(),
    setNewTextField:jest.fn()
  };

  it('should show empty textfield  when component is rendered',()=>{
    expect(screen.queryByTestId('newConfig-key')).toBeFalsy();
    render(<CustomTextField userConfig = {mock.userConfig} handleUserConfig={mock.handleUserConfig} setNewTextField = {mock.setNewTextField}/>);
    expect(screen.queryByTestId('newConfig-key')).toBeTruthy();
    expect(screen.queryByTestId('newConfig-key').value).toEqual(undefined);
  });

  it('should update newConfigKey when textfield is updated',()=>{
    expect(screen.queryByTestId('newConfig-key')).toBeFalsy();
    render(<CustomTextField userConfig = {mock.userConfig} handleUserConfig={mock.handleUserConfig} setNewTextField = {mock.setNewTextField}/>);
    expect(screen.queryByTestId('newConfig-key')).toBeTruthy();
    expect(screen.queryByLabelText('Enter Key')).toBeTruthy();
    const configKey = screen.getByTestId('newConfig-key').querySelector('input');
    fireEvent.change(configKey,{target:{value:'name'}});
    expect(screen.queryByDisplayValue('name')).toBeTruthy();
  });

  it('should show add config button when value-textfield has some value',()=>{
    expect(screen.queryByTestId('newConfig-value')).toBeFalsy();
    render(<CustomTextField userConfig = {mock.userConfig} handleUserConfig={mock.handleUserConfig} setNewTextField = {mock.setNewTextField}/>);
    expect(screen.queryByLabelText('Enter Value')).toBeTruthy();
    const configKey = screen.getByTestId('newConfig-key').querySelector('input');
    fireEvent.change(configKey,{target:{value:'name'}});
    expect(screen.queryByTestId('newConfig-value')).toBeTruthy();
    const configValue = screen.queryByTestId('newConfig-value').querySelector('input');
    fireEvent.change(configValue,{target:{value:'2020'}});
    expect(screen.queryByDisplayValue('2020')).toBeTruthy();

  });


  it('should render correctly and create a snapshot', () => {
    const {asFragment } =  render(<CustomTextField userConfig = {mock.userConfig} handleUserConfig={mock.handleUserConfig} setNewTextField = {mock.setNewTextField}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});