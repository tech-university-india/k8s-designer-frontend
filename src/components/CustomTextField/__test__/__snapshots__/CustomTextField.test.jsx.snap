// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frontend Configuration should render correctly and create a snapshot 1`] = `
<DocumentFragment>
  <form
    autocomplete="off"
    class="custom-textfield-container config-layout MuiBox-root css-19krtbh"
    novalidate=""
    style="font-family: Open Sans;"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      data-testid="newConfig-key"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        Enter Key
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="outlined-basic"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Enter Key
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      data-testid="newConfig-value"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        Enter Value
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="outlined-basic"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Enter Value
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
